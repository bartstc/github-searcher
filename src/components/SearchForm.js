import React from 'react';

const SearchForm = ({ onChange, fetchRepos }) => (
  <>
    <div className="input-group mb-3">
      <input onChange={onChange} type="text" className="search-input form-control" placeholder="Search for repo ..." aria-label="Recipient's username" aria-describedby="basic-addon2" />
      <div className="input-group-append">
        <button className="search-submit btn btn-outline-secondary" type="button" onClick={fetchRepos}>Search</button>
      </div>
    </div>
  </>
);

export default SearchForm;