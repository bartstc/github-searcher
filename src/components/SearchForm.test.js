import React from 'react';
import { shallow, mount } from 'enzyme';
import toJson from 'enzyme-to-json';
import SearchForm from './SearchForm';

describe('SearchForm', () => {
  let form = shallow(<SearchForm />);

  it('renders properly', () => {
    expect(toJson(form)).toMatchSnapshot();
  });

  describe('when submit', () => {
    const mockFetchRepos = jest.fn();

    beforeEach(() => {
      const props = { fetchRepos: mockFetchRepos };
      form = mount(<SearchForm {...props} />);

      form.find('.search-submit').simulate('click');
    });

    it('dispatches the `fetchRepos()` method it receives from props', () => {
      expect(mockFetchRepos).toHaveBeenCalled();
    });
  });
});