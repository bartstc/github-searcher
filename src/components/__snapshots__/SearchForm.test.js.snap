// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SearchForm renders properly 1`] = `
<Fragment>
  <div
    className="input-group mb-3"
  >
    <input
      aria-describedby="basic-addon2"
      aria-label="Recipient's username"
      className="search-input form-control"
      placeholder="Search for repo ..."
      type="text"
    />
    <div
      className="input-group-append"
    >
      <button
        className="search-submit btn btn-outline-secondary"
        type="button"
      >
        Search
      </button>
    </div>
  </div>
</Fragment>
`;
